app/Http/Controllers/Licenses/LicenseCheckoutController.php

```diff
@@ -126,4 +126,65 @@ protected function checkoutToUser($licenseSeat)
 
         return false;
     }
+
+    /**
+     * Bulk checkin all license seats
+     *
+     * @author [A. Gianotto] [<snipe@snipe.net>]
+     * @see LicenseCheckinController::create() method that provides the form view
+     * @since [v6.1.1]
+     * @return \Illuminate\Http\RedirectResponse
+     * @throws \Illuminate\Auth\Access\AuthorizationException
+     */
+
+    public function bulkCheckout($licenseId) {
+
+        \Log::debug('checking out '.$licenseId.' via bulk');
+        $license = License::findOrFail($licenseId);
+        $this->authorize('checkin', $license);
+
+        $users = User::whereNull('deleted_at')->where('autoassign_licenses', '=', 1)->with('licenses')->get();
+        \Log::debug($users->count().' will be assigned');
+
+        if ($users->count() > $license->getAvailSeatsCountAttribute()) {
+            \Log::debug('You do not have enough free seats to complete this task, so we will check out as many as we can. ');
+        }
+
+        // If the license is valid, check that there is an available seat
+        if ($license->availCount()->count() < 1) {
+            return redirect()->back()->with('error', 'No more available seats');
+        }
+
+        $avail_count = $license->availCount()->count();
```

Comment:
Minor nit: as opposed to saying this twice, you can hoist this line above the `if()` and then in the `if()` just check against `$avail_count`.